lib/github.js

```diff
@@ -1,62 +1,110 @@
 /* eslint-disable camelcase */
 
-var base = 'https://api.github.com/';
+/**
+ * Module dependencies.
+ */
+
+var debug = require('debug')('duo:gh-resolve:github');
+var request = require('co-request');
 var path = require('path');
-var request = require('request');
 
-module.exports = function (opts) {
-  var token = opts.token || opts.password;
 
-  return {
-    tags: tags.bind(null, token),
-    branch: branch.bind(null, token)
-  };
+/**
+ * Retrieve the list of tags for a given repository.
+ *
+ * @param {String} token
+ * @param {Object} options
+ * @return {Object}
+ */
+
+exports.tags = function* (token, options) {
+  return yield get(token, [
+    'repos',
+    options.owner,
+    options.repo,
+    'tags'
+  ]);
 };
 
-function tags(token, owner, repo, callback) {
-  return request({
-    url: url('repos', owner, repo, 'tags'),
-    qs: { access_token: token },
-    json: true,
-    headers: {
-      'Accept': 'application/vnd.github.quicksilver-preview+json',
-      'User-Agent': 'duo:gh-resolve'
-    },
-    followRedirects: true
-  }, handle(callback));
-}
+/**
+ * Retrieve the information about a specific branch for a repository.
+ *
+ * @param {String} token
+ * @param {Object} options
+ * @return {Object}
+ */
+
+exports.branch = function* (token, options) {
+  return yield get(token, [
+    'repos',
+    options.owner,
+    options.repo,
+    'branches',
+    options.branch
+  ]);
+};
+
+/**
+ * Helper for performing an HTTP GET against the Github API.
+ *
+ * @param {String} token
+ * @param {Array:String} parts  Pieces of the URL
+ * @return {Response}
+ */
+
+function* get(token, parts, attempts) {
+  var url = 'https://api.github.com/' + path.join.apply(path, parts);
```

Comment:
Oh snap... you're probably right. We should use `url.resolve` instead anyways. Good catch.